src/node_options.cc

```diff
@@ -128,7 +128,12 @@ void PerIsolateOptions::HandleMaxOldSpaceSizePercentage(
   }
 
   // Get available memory in bytes
+#ifdef V8_COMPRESS_POINTERS
+  // When pointer compression is enabled, V8 uses a 4 GiB heap limit.
+  uint64_t total_memory = 4294967296;  // 4 GiB
```

Comment:
This is a bit more readable (or declare it as a constant)

```suggestion
  uint64_t total_memory = uint64_t{1} << 32;  // 4 GiB
```